tests/integration/conftest.py

```diff
@@ -101,21 +102,30 @@ def fixture_application_name():
 
 
 @pytest_asyncio.fixture(scope="function", name="default_admin_password")
-async def fixture_default_admin_password(ops_test: pytest_operator.plugin.OpsTest):
-    return_code, stdout, stderr = await ops_test.juju(
-        "run-action",
-        "wordpress/0",
-        "get-initial-password",
-        "--wait",
-        "--format",
-        "json"
-    )
-    assert (
-            return_code == 0 and not stderr
-    ), "juju run-action get-initial-password should not fail"
-    result = json.loads(stdout)["unit-wordpress-0"]
-    assert (
-            result["status"] == "completed"
-    ), "get-initial-password action should have a status of completed"
-
-    yield result["results"]["password"]
+async def fixture_default_admin_password(
+        ops_test: pytest_operator.plugin.OpsTest,
+        application_name
+):
+    application: juju.application = ops_test.model.applications[application_name]
+    action: juju.action.Action = await application.units[0].run_action("get-initial-password")
+    await action.wait()
+
+    yield action.results["password"]
+
+
+@pytest_asyncio.fixture(scope="function", name="get_unit_ip_list")
+async def fixture_get_unit_ip_list(
+        ops_test: pytest_operator.plugin.OpsTest,
+        application_name
+):
+    """Helper function to retrieve unit ip addresses, similar to fixture_get_unit_status_list"""
+
+    async def _get_unit_ip_list():
+        status = await ops_test.model.get_status()
+        units = status.applications[application_name].units
+        ip_list = []
+        for key in sorted(units.keys(), key=lambda n: int(n.split("/")[-1])):
+            ip_list.append(units[key].address)
+        return ip_list
+
+    return _get_unit_ip_list
```

Comment:
Nice work moving this to a fixture, could this fixture just return the value of the `_get_unit_ip_list` rather than returning the function?

Reply:
In some of the future test cases, the unit replicas number could change in the test case (for testing charm behavior after add-unit), but I think I can add another fixture named just `unit_ip_list` for getting the IP address directly for test cases without unit number changing.